auction-server/src/auction.rs

```diff
@@ -625,6 +630,60 @@ pub struct Bid {
     pub amount:          BidAmount,
 }
 
+#[serde_as]
+#[derive(Serialize, Deserialize, ToSchema, Clone, Debug)]
+pub struct BidSvm {
+    /// The permission key to bid on.
+    #[schema(example = "SGVsbG8sIFdvcmxkIQ==", value_type = String)]
+    #[serde_as(as = "Base64")]
+    pub permission_key: Bytes,
+    /// The chain id to bid on.
+    #[schema(example = "solana", value_type = String)]
+    pub chain_id:       ChainId,
+    /// Bid amount in lamports.
+    #[schema(example = 10, value_type = u64)]
+    pub amount:         u64,
+    /// The transaction for bid.
+    #[schema(example = "SGVsbG8sIFdvcmxkIQ==", value_type = String)]
+    #[serde_as(as = "Base64")]
+    pub transaction:    Vec<u8>,
+}
+
+#[derive(Serialize, ToSchema, Debug, Clone)]
+#[serde(untagged)] // Remove tags to avoid key-value wrapping
+pub enum Bid {
+    Evm(BidEvm),
+    Svm(BidSvm),
+}
+
+impl<'de> Deserialize<'de> for Bid {
+    fn deserialize<D>(d: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        struct BidChainId {
+            chain_id: ChainId,
+        }
+
+        let value: serde_json::Value = Deserialize::deserialize(d)?;
+        let bid_id: BidChainId =
+            serde_path_to_error::deserialize(&value).map_err(serde::de::Error::custom)?;
+        match bid_id.chain_id.as_str() {
+            "solana" => {
+                let svm_bid: BidSvm =
+                    serde_path_to_error::deserialize(&value).map_err(serde::de::Error::custom)?;
```

Comment:
just out of curiosity, is this a good enough error to return to users?

Reply:
This will be similar to what we had previously. as an example the message would be like:
"Failed to deserialize the JSON body into the target type: amount: invalid type: string "10", expected u64"

To ensure that we have the field that caused the error inside the message, I used "serde_path_to_error". Since we are deserializing a JSON variable, the field would not be in the error message without such a library.